### ISSUE TITLE
Transfer learning

### ISSUE THREAD

**Issue by Saiflalwani** [ultralytics/ultralytics#2339]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

i have trained yolo v8 model for englishcards having classes ('Smart Identifier',' Smart Back', 'Logo','Picture',' Smart Front','Damaged')

this is the yaml file
train: ../train/images
val: ../valid/images
test: ../test/images

nc: 6
names: [' Smart Identifier', 'Smart Back', ' Logo', 'Picture', 'Smart Front', 'Damaged']

now i have seprately annotated urducard having classes ('urdu Identifier',' urduBack', 'Logo','Picture',' urduFront') the sequence of annotation ('urdu Identifier',' urduBack', 'Logo','Picture',' urduFront') starting from 0 index
so urdu have 5 classes i whant to consolidate these twoenglishcard and urducard i both models Logo and picture are same

now i have separately annotates urdu sets containg urdu Identifier',' urduBack', 'Logo','Picture',' urduFront'
now i want apply transfer learning so i can make a consolidate model

i have created new yaml file added all the classes 
train: ../train/images
val: ../valid/images
test: ../test/images

nc: 11
names: [ ' Urdu Identifier', ' Urdu Back', 'Logo', 'Picture' ,' Urdu Front'' Identifier' ,'Smart Back', ' Logo', 'Picture' ,'Smart Front', 'Damaged']

i have uses this code for training
!yolo task=detect mode=train model=/content/detect/train/weights/best.pt data=data.yaml epochs=100 batch=16 imgsz=650 pretrained=True

model=/content/detect/train/weights/best.pt
this is the model that is trained on englishcards
in data.yaml training images contain only urdu datasets that is annotated on urdu classes

what things to be updated in training and yaml file what else should i do to complete this task to make a consolidate model to detect all the classes with same model

@AyushExel  @plashchynski @CySlider  @akx  @pkehrer  @onuralpszr

**Comment by AyushExel** [comment]
The dataset you're training on should be a complete dataset containing files for all classes that you intend to train on. You cannot retrain the previously trained classes on the model and re-train only on new classes. Transfer learning allows you to train a pre-trained model on a NEW dataset but it doesn't retain the classes of previous dataset

**Comment by TapendraBaduwal** [comment]
@AyushExel 

i am confused Lets i have coco128.yaml file in below way..

path: ../datasets/coco128 # dataset root dir
train: images/train2017 # train images (relative to 'path') 128 images
val: images/train2017 # val images (relative to 'path') 128 images
test: # test images (optional)

names:
0: person
1: bicycle
2: car
3: motorcycle
4: airplane
5: bus
.........

But after annotated i got this one file
train: ../train/images
val: ../valid/images
test: ../test/images

nc: 3
names: ['bus', 'person', 'laptop']

I am going to fine tune existing model while fine tune i have to give path as follow
results = model.train(data='newdata.yaml', epochs=3) # train the model

Here the new annotation bus and person label same as existing and lets "laptop" is new label

Now in which file we have to add new label "laptop" because i am doing finetune in new data.yaml ..we are not using coco128.yaml file while finetune?

In new annotation we didi not get id name of label we just get
nc: 3
names: ['bus', 'person', 'laptop']

while finetune how model map my new label and old labels ?

**Comment by AyushExel** [comment]
Your new file should work. The resultant model will have 3 classes - ['bus', 'person', 'laptop']

**Comment by TapendraBaduwal** [comment]
@AyushExel But i want to apply Transfer learning and want to fine tune existing model....My new model must predict old class label and new class label...how can i do this?

**Comment by AyushExel** [comment]
We don't have any officially tested recipes for that type of transfer learning where you preserve the classes learned from the previous dataset. Remember, fine-tuning is a method of transfer learning.
What you're trying to achieve is more nuanced. You'll need to freeze some layers of the pre-trained model and then tune some hyper-params to prevent the model from forgetting the previous dataset. 
And secondly, the dataset format doesn't really support transfer learning from any previous dataset.

A better solution would be to just mix your new dataset with "m" classes and your old dataset with "n" set form a new dataset with set(m+n) classes and then train your model on that dataset

**Comment by TapendraBaduwal** [comment]
@AyushExel Dear AyushExel I want to develop a system in which if any FP and FN data we found then just annotated these data and want to train existing model with new FP and FN data so our new model just contineously learned from mistake....

Does below code help to solve this...
model = YOLO('yolov8n.yaml')  # build a new model from scratch
model = YOLO('yolov8n.pt')  # load a pretrained model (recommended for training)

model = YOLO('yolov8n.yaml').load('yolov8n.pt')  # build from YAML and transfer weights

results = model.train(data='newdata.yaml', epochs=3)  # train the model
